test/Algebra/Graph/Test/Undirected.hs

```diff
@@ -0,0 +1,43 @@
+-----------------------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Test.Undirected
+-- Copyright  : (c) Andrey Mokhov 2016-2019
+-- License    : MIT (see the file LICENSE)
+-- Maintainer : andrey.mokhov@gmail.com
+-- Stability  : experimental
+--
+-- Testsuite for "Algebra.Graph.Undirected".
+-----------------------------------------------------------------------------
+module Algebra.Graph.Test.Undirected (
+    -- * Testsuite
+    testUndirected
+    ) where
+
+import Algebra.Graph.Undirected
+import Algebra.Graph.Test
+import Algebra.Graph.Test.API (toIntAPI, undirectedGraphAPI)
+import Algebra.Graph.Test.Generic
+
+tPoly :: Testsuite Graph Ord
+tPoly = ("Graph.Undirected.", undirectedGraphAPI)
+
+t :: TestsuiteInt Graph
+t = fmap toIntAPI tPoly
+
+type G = Graph Int
+
+testUndirected :: IO ()
+testUndirected = do
+    putStrLn "\n============ Graph.Undirected ============"
+    test "Axioms of undirected graphs" $
+        size10 (undirectedAxioms :: GraphTestsuite G)
```

Comment:
Compared to `SymmetricRelation`, we miss some tests:

https://github.com/snowleopard/alga/blob/master/test/Algebra/Graph/Test/Relation/SymmetricRelation.hs

For example, `fromUndirected` is not tested. I guess we need to test it. 

Do we need to add `toUndirected` too, for consistency of the API?

Reply:
On it! I think it makes sense to provide `toUndirected` it wont hurt since it will be just a coercion, right?

Reply:
Actually, I don't think that `toUndirected` and `fromUndirected` should work as `toSymmetric` and  `fromSymmetric` since we decided to convert it to a normal `Graph` without redundant edges. Some of the properties will be different, is that okay?